commons/src/main/java/com/powsybl/commons/binary/BinReader.java

```diff
@@ -76,7 +76,8 @@ private String readString() {
             }
             byte[] stringBytes = dis.readNBytes(stringNbBytes);
             if (stringBytes.length != stringNbBytes) {
-                throw new PowsyblException("Cannot read the full string, bytes missing: " + (stringNbBytes - stringBytes.length));
+                throw new PowsyblException("Cannot read the full string, bytes missing: " + (stringNbBytes - stringBytes.length)
+                        + " (this may happen when the attribute wasn't written in the first place, causing string length to be an aberrant number)");
```

Comment:
This should not be reported in the exception. But it would be useful to keep it as a comment in the code.